ckan/logic/action/update.py

```diff
@@ -123,6 +125,8 @@ def resource_update(context, data_dict):
     upload.upload(id, uploader.get_max_resource_size())
     model.repo.commit()
```

Comment:
The main thing that should happen is for other transactions to wait for the lock to be released. Quoting the postgres documentation: "_FOR UPDATE causes the rows retrieved by the SELECT statement to be locked as though for update. This prevents them from being modified or deleted by other transactions until the current transaction ends. That is, other transactions that attempt UPDATE, DELETE, or SELECT FOR UPDATE of these rows will be blocked until the current transaction ends_."

There is also this in the documentation "_Within a SERIALIZABLE transaction, however, an error will be thrown if a row to be locked has changed since the transaction started_" BUT I think this is only the case when postgres is set to use the "SERIALIZABLE" transaction isolation level.